### ISSUE TITLE
Version 1.4.4 causing "promise rejection" error - no longer working with ConBee II

### ISSUE THREAD

**Issue by lansester** [ioBroker/ioBroker.zigbee#1000]
Hi guys,

I am using a ConBee II adapter since a longer while and after installing Version 1.4.4 a few days ago I recognised error messages in my IoBroker log.
They came again and again and always started with 
"promise rejection. This error originated either by throwing inside of an async function"

After falling back to V 1.4.1 all was working properly again with no errors/warnings or so.
Hope you can fix it.
Thanks a lot!

**Comment by Apollon77** [comment]
Please add a full log ... not only one line

**Comment by lansester** [comment]
The following appears every 30secs in the log:

========================================

2021-02-15 11:52:20.357  - [32minfo[39m: zigbee.0 (41874) starting. Version 1.4.4 in /opt/iobroker/node_modules/iobroker.zigbee, node: v12.17.0, js-controller: 3.1.6
2021-02-15 11:52:20.393  - [31merror[39m: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-02-15 11:52:20.395  - [31merror[39m: zigbee.0 (41874) unhandled promise rejection: PanID must have a value of 0x0001 (1) - 0xFFFE (65534), got 20111210.
2021-02-15 11:52:20.397  - [31merror[39m: zigbee.0 (41874) Error: PanID must have a value of 0x0001 (1) - 0xFFFE (65534), got 20111210.
    at new Controller (/opt/iobroker/node_modules/zigbee-herdsman/dist/controller/controller.js:95:19)
    at ZigbeeController.configure (/opt/iobroker/node_modules/iobroker.zigbee/lib/zigbeecontroller.js:79:25)
    at Zigbee.onReady (/opt/iobroker/node_modules/iobroker.zigbee/main.js:142:27)
    at Zigbee.emit (events.js:315:20)
    at /opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:8148:34
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
2021-02-15 11:52:20.398  - [32minfo[39m: zigbee.0 (41874) cleaned everything up...
2021-02-15 11:52:20.400  - [33mwarn[39m: zigbee.0 (41874) Failed to stop zigbee during startup
2021-02-15 11:52:20.401  - [31merror[39m: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-02-15 11:52:20.401  - [31merror[39m: zigbee.0 (41874) unhandled promise rejection: Cannot read property 'database' of undefined
2021-02-15 11:52:20.402  - [31merror[39m: zigbee.0 (41874) TypeError: Cannot read property 'database' of undefined
    at ZigbeeController.getClients (/opt/iobroker/node_modules/iobroker.zigbee/lib/zigbeecontroller.js:205:27)
    at DeviceEvent.stop (/opt/iobroker/node_modules/iobroker.zigbee/lib/zbDeviceEvent.js:25:25)
    at ZigbeeController.callExtensionMethod (/opt/iobroker/node_modules/iobroker.zigbee/lib/zigbeecontroller.js:192:42)
    at ZigbeeController.stop (/opt/iobroker/node_modules/iobroker.zigbee/lib/zigbeecontroller.js:310:20)
    at Zigbee.onUnload (/opt/iobroker/node_modules/iobroker.zigbee/main.js:581:41)
    at Zigbee.emit (events.js:315:20)
    at stop (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:8270:22)
    at exceptionHandler (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:8329:13)
    at process.<anonymous> (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:8342:45)
    at process.emit (events.js:315:20)
2021-02-15 11:52:20.403  - [31merror[39m: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-02-15 11:52:20.404  - [31merror[39m: zigbee.0 (41874) unhandled promise rejection: Cannot read property 'database' of undefined
2021-02-15 11:52:20.404  - [31merror[39m: zigbee.0 (41874) TypeError: Cannot read property 'database' of undefined
    at ZigbeeController.getClients (/opt/iobroker/node_modules/iobroker.zigbee/lib/zigbeecontroller.js:205:27)
    at DeviceAvailability.stop (/opt/iobroker/node_modules/iobroker.zigbee/lib/zbDeviceAvailability.js:222:43)
    at ZigbeeController.callExtensionMethod (/opt/iobroker/node_modules/iobroker.zigbee/lib/zigbeecontroller.js:192:42)
    at ZigbeeController.stop (/opt/iobroker/node_modules/iobroker.zigbee/lib/zigbeecontroller.js:310:20)
    at Zigbee.onUnload (/opt/iobroker/node_modules/iobroker.zigbee/main.js:581:41)
    at Zigbee.emit (events.js:315:20)
    at stop (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:8270:22)
    at exceptionHandler (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:8329:13)
    at process.<anonymous> (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:8342:45)
    at process.emit (events.js:315:20)
2021-02-15 11:52:20.549  - [32minfo[39m: zigbee.0 (41874) terminating
2021-02-15 11:52:20.550  - [32minfo[39m: zigbee.0 (41874) Terminated (NO_ERROR): Without reason
2021-02-15 11:52:21.069  - [32minfo[39m: host.HomeServer instance system.adapter.zigbee.0 terminated with code 0 (NO_ERROR)
2021-02-15 11:52:21.069  - [32minfo[39m: host.HomeServer Restart adapter system.adapter.zigbee.0 because enabled

========================================

**Comment by asgothian** [comment]
You seem to have an error in your settings. The log actually tells you about it:

Error: PanID must have a value of 0x0001 (1) - 0xFFFE (65534), got 20111210.


Please check your adapter settings and ensure that the PanID is set to a correct value.

Note that this is not caused by a change in the zigbee adapter but by a change in the underlying zigbee library.

A.

**Comment by lansester** [comment]
Thanks a lot for a response!

Well, if I would have an error in my settings, e.g. the chosen PanID is not within the allowed range, I would have expected, that the adapter at least explains the valid range within the description next to the PanID field.
Or that the adapter declines to accept any invalid value.
But neither was the case when I initially defined those values. There was no hint for the valid range and the adapter accepted my chosen value.
Therefore I do not think that I have made any error when initlally configuring the change.
Either the adapter had a gap or bug to accept invalid values or the allowed value range has changed since then.
In case the range for valid values got changed, I would see that as a disruptive change which requires a heads-up announcement of several months at least as it completely disables existing zigbee configurations with a PanID outside of the newly introduced limits.
I thought that the aim of IoBroker is to be a bit more than just a tinker playground and should also be used in productive environments. But that also needs a professional handling of major/disruptive changes which seems to not have happened here. 
Unfortunately the zigbee adapter also tells nothing about the impact of a PanID change.
In case it requires to pair all devices again it is an absolute no-go which I can not afford from a timing perspective.
I was forced to switch from the Lightify adapter to this one as Lightify was discontinued even 1,5 years before the actual Osram platform is closed which was way to early in my view. 
And now the next major issue with this implementation.....
Just my honest view from a user perspective.
No criticism about the massive work invested in the IoBroker platform which is really impressive.
But such cases like this show, that in the end IoBroker is probably not yet what it wants to be "afaik" (a professional implementation and not just a playground).
However, big thanks again for your response and the hint towards the root cause.
